compiler/AST/expr.cpp

```diff
@@ -3722,9 +3722,67 @@ CallExpr::insertAtTail(BaseAST* ast) {
 }
 
 
-FnSymbol* CallExpr::isResolved(void) {
-  SymExpr* base = toSymExpr(baseExpr);
-  return base ? toFnSymbol(base->var) : NULL;
+bool CallExpr::isEmpty() const {
+  return primitive == NULL && baseExpr == NULL;
+}
+
+
+// MDN 2016/01/29: This will become a predicate
+FnSymbol* CallExpr::isResolved() const {
+  return resolvedFunction();
+}
+
+
+FnSymbol* CallExpr::resolvedFunction() const {
```

Comment:
You mean resolvedFunction()?

It is true that resolvedFunction has been written in a manner that might do "some repeated work" esp relative
to all of the current calls to isResolved(). However I think it is unlikely that the impact of any such work is
measurable in the current compiler.

In the immediate term it helps to clarify these cases in which isResolved() is returning NULL.
In the middle term it helps to provide some cheap validation as existing uses of isResolved() are being revised.
In the long term the function might be streamlined a bit.